Frameworks/CoreImage/CIImage.mm

```diff
@@ -73,90 +74,164 @@ @implementation CIImage
 /**
  @Status Interoperable
 */
-- (CIImage*)init {
++ (CIImage*)emptyImage {
+    return [[[CIImage alloc] init] autorelease];
+}
+
+/**
+ @Status Interoperable
+*/
+- (CGRect)extent {
+    return self->_extent;
+}
+
+/**
+ @Status Interoperable
+*/
+- (instancetype)init {
     if (self = [super init]) {
-        _cgImage = nil;
-        _color = nil;
-        _filter = nil;
         _extent = CGRectInfinite;
     }
     return self;
 }
 
+- (instancetype)_initWithCGImage:(CGImageRef)image color:(CIColor*)color {
+    if (self = [self initWithCGImage:image]) {
+        _color.attach([static_cast<id>(color) copy]);
+    }
+
+    return self;
+}
+
+/**
+ @Status Interoperable
+*/
+- (instancetype)initWithColor:(CIColor*)color {
+    if (!color) {
+        [self release];
+        return nil;
+    }
+
+    if (self = [self init]) {
+        _color.attach([static_cast<id>(color) copy]);
+    }
+
+    return self;
+}
+
+/**
+ @Status Interoperable
+*/
+- (instancetype)initWithCGImage:(CGImageRef)cgImage {
+    if (!cgImage) {
+        [self release];
+        return nil;
+    }
+
+    if (self = [self init]) {
+        _cgImage.reset(CGImageRetain(cgImage));
+        _extent = CGRectMake(0, 0, CGImageGetWidth(cgImage), CGImageGetHeight(cgImage));
+    }
+
+    return self;
+}
+
 /**
  @Status Stub
  @Notes
 */
-+ (CIImage*)emptyImage {
+- (instancetype)initWithCGImage:(CGImageRef)image options:(NSDictionary*)d {
     UNIMPLEMENTED();
     return StubReturn();
 }
 
 /**
  @Status Interoperable
 */
-+ (CIImage*)imageWithColor:(CIColor*)color {
-    CIImage* ret = [[CIImage alloc] init];
-    if (ret != nil) {
-        ret->_color.attach([static_cast<id>(color) copy]);
-        ret->_extent = CGRectMake(INFINITY, INFINITY, CGFLOAT_MAX, CGFLOAT_MAX);
+- (instancetype)initWithImage:(UIImage*)image {
+    if (!image) {
+        [self release];
+        return nil;
     }
-    return [ret autorelease];
+    return [self initWithCGImage:[image CGImage]];
 }
 
 /**
  @Status Stub
  @Notes
 */
-+ (CIImage*)imageWithBitmapData:(NSData*)d bytesPerRow:(size_t)bpr size:(CGSize)size format:(CIFormat)f colorSpace:(CGColorSpaceRef)cs {
+- (instancetype)initWithImage:(UIImage*)image options:(NSDictionary*)options {
```

Comment:
It'd be quick to move the code from initWithImage: into this function and add `@Notes` saying that options is unsupported. Then you'd move one API out of Stub and into Caveat!